programs/mpl-core/src/plugins/lifecycle.rs

```diff
@@ -857,3 +859,68 @@ pub(crate) fn validate_plugin_checks<'a>(
         Ok(ValidationResult::Pass)
     }
 }
+
+/// This function iterates through all external plugin checks passed in and performs the validation
+/// by deserializing and calling validate on the plugin.
+/// The STRONGEST result is returned.
+#[allow(clippy::too_many_arguments, clippy::type_complexity)]
+pub(crate) fn validate_external_plugin_checks<'a>(
+    key: Key,
+    external_checks: &BTreeMap<
+        ExternalPluginKey,
+        (Key, ExternalCheckResultBits, ExternalRegistryRecord),
+    >,
+    authority: &'a AccountInfo<'a>,
+    new_owner: Option<&'a AccountInfo<'a>>,
+    new_plugin: Option<&Plugin>,
+    asset: Option<&AccountInfo<'a>>,
+    collection: Option<&AccountInfo<'a>>,
+    resolved_authorities: &[Authority],
+    external_plugin_validate_fp: fn(
+        &ExternalPlugin,
+        &PluginValidationContext,
+    ) -> Result<ValidationResult, ProgramError>,
+) -> Result<ValidationResult, ProgramError> {
+    let mut approved = false;
+    let mut rejected = false;
+    for (check_key, check_result, external_registry_record) in external_checks.values() {
+        if *check_key == key
+            && (check_result.can_listen()
+                || check_result.can_approve()
+                || check_result.can_reject())
+        {
+            let account = match key {
+                Key::CollectionV1 => collection.ok_or(MplCoreError::InvalidCollection)?,
+                Key::AssetV1 => asset.ok_or(MplCoreError::InvalidAsset)?,
+                _ => unreachable!(),
+            };
+
+            let ctx = PluginValidationContext {
+                self_authority: &external_registry_record.authority,
+                authority_info: authority,
+                resolved_authorities: Some(resolved_authorities),
+                new_owner,
+                target_plugin: new_plugin,
+            };
+
+            let result = external_plugin_validate_fp(
+                &ExternalPlugin::load(account, external_registry_record.offset)?,
+                &ctx,
+            )?;
+            match result {
+                ValidationResult::Rejected => rejected = true,
+                ValidationResult::Approved => approved = true,
+                ValidationResult::Pass => continue,
+                ValidationResult::ForceApproved => return Ok(ValidationResult::ForceApproved),
```

Comment:
I think we can chuck force approve and simplify things more.

Reply:
So agree I should remove `ForceApproved` here because we won't have any external validations return it.  However, I don't know if we want to make a new `PluginValidation` trait for external plugins (because its a lot more copy and paste).

So in that case, the functions that respect the original `PluginValidation` trait return a plain `ValidationResult` instead of an `ExternalValidationResult`.  So I have to handle the case in the match statement.

So the easy way to chuck it is like:
`ValidationResult::Pass | ValidationResult::ForceApproved => continue`.

The other way to remove it is with types: to either make an `ExternalValidationPlugin` trait, or just modify the `ExternalPlugin` level functions to return `ExternalValidationResult` even if the trait functions return `ValidationResult`.

For now just "chucking" a returned `ForceApproved`, lmk what you think.

Reply:
I'm fine with the chucking approach. I mostly just wanted a fast exit if encountering a Rejection.

Reply:
ah good idea.  Fixed and also marked `ForceApproved` as unreachable so that it would panic as its not an expected case unless we introduce a code bug.